File: frontend/code/models/creature.py
from dataclasses import dataclass

from models.stats import Stats


@dataclass
class Creature:
    name: str
    stats: Stats
    totalSpeed: float
    abilities: object = None # dictionary of abilities, display name: id

    @classmethod
    def from_dict(cls, d):
        return cls(
            name=d['name'],
            stats=Stats.from_dict(d['stats']),
            abilities=d.get('abilities') or {},
            totalSpeed=d['totalSpeed'],
        )

File: front-end/src/models/Creature.js
import { Stats } from './Stats.js';

export class Creature {
    constructor(name, stats, totalSpeed, abilities) {
        this.name = name;
        this.stats = stats;
        this.totalSpeed = totalSpeed;
        this.abilities = abilities;
    }

    static fromDict(data) {
        return new Creature(
            data.name,
            data.stats ? Stats.fromDict(data.stats) : null,
            data.totalSpeed,
            data.abilities
        );
    }
}
